File: funnyvote/app/src/main/java/com/heaton/funnyvote/MainPageContract.java
package com.heaton.funnyvote;

import com.heaton.funnyvote.database.User;

public interface MainPageContract {
    interface Presenter extends BasePresenter {
        void IntentToSearchPage(String searchKeyword);

        void IntentToCreatePage();

        void IntentToUserPage();

        void IntentToMainPage();

        void IntentToAboutPage();

        void IntentToAccountPage();

        void loadUser();
    }

    interface View extends BaseView<Presenter> {
        void showSearchPage(String searchKeyword);

        void showCreatePage();

        void showUserPage();

        void showMainPage();

        void showAboutPage();

        void showAccountPage();

        void updateUserView(User user);
    }
}

File: funnyvote/app/src/main/java/com/heaton/funnyvote/MainPageContract.kt
package com.heaton.funnyvote

import com.heaton.funnyvote.database.User

interface MainPageContract {
    interface Presenter : BasePresenter {
        fun IntentToSearchPage(searchKeyword: String)

        fun IntentToCreatePage()

        fun IntentToUserPage()

        fun IntentToMainPage()

        fun IntentToAboutPage()

        fun IntentToAccountPage()

        fun loadUser()
    }

    interface View : BaseView<Presenter> {
        fun showSearchPage(searchKeyword: String)

        fun showCreatePage()

        fun showUserPage()

        fun showMainPage()

        fun showAboutPage()

        fun showAccountPage()

        fun updateUserView(user: User)
    }
}
